### PR TITLE
valent: init at unstable-2023-03-02

### LINKED ISSUE #218392: Package request: Valent
**Project description**

Valent is an implementation of the KDE Connect protocol, built on GNOME platform libraries.

Seems to be a re-implementation of [GSConnect](https://github.com/GSConnect/gnome-shell-extension-gsconnect) without the GNOME Shell bits made by the same author, intended to work on more desktop environments but it also has a (separate) GNOME Shell extension.

I have packaged it myself ([here](https://github.com/FedericoSchonborn/valent-nix/blob/main/packages/valent/default.nix)) and it works surprisingly well for an alpha.

imagen

**Metadata**

* homepage URL: https://valent.andyholmes.ca/
* source URL: https://github.com/andyholmes/valent
* license: gpl3Plus, cc0
* platforms: linux

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

Valent is an implementation of the KDE Connect protocol, built on GNOME platform libraries.

This patch resolves #218392.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by foo-dogsquared** [comment]
Result of `nixpkgs-review pr 218401` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>valent</li>
  </ul>
</details>

**Comment by foo-dogsquared** [comment]
Similar results from [my previous comment on the linked issue](https://github.com/NixOS/nixpkgs/issues/218392#issuecomment-1445290087). Most of the features I use work here so I think looks good to me.

**Review by foo-dogsquared — APPROVED** [review]

File: pkgs/applications/misc/valent/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "valent";
+  version = "unstable-2023-02-27";
+
+  src = fetchFromGitHub {
+    owner = "andyholmes";
+    repo = "valent";
+    rev = "e35147fd334b9dc5e3ae71f5943fd94286611a4a";
+    fetchSubmodules = true;
```
[the remark this answers is not in the document]
Comment (FedericoSchonborn) [reply pkgs/applications/misc/valent/default.nix:29]: > libgnome-volume-control is a copy library that's supposed to be used as
a git sub-module.

https://github.com/search?q=libgnome-volume-control+path%3A.gitmodules&type=code

**Comment by Aleksanaa** [comment]
I was testing https://github.com/andyholmes/valent/tree/401fda881bc17d2424a621d41c9345f5e9e4f689 and saw that there's a dependency check of `libsysprof-capture`, although that was a non-blocker.

**Pull request merged**